plugins/uploader-resources/src/store.ts

```diff
@@ -14,48 +14,45 @@
 //
 
 import { type FileUploadTarget } from '@hcengineering/uploader'
-
-import { type Uppy } from '@uppy/core'
 import { type Writable, writable } from 'svelte/store'
 
 /** @public */
-export interface FileUpload {
-  target: FileUploadTarget
-  uppy: Uppy
+export interface Upload {
+  progress: number
+  error?: string
+  retry?: () => Promise<void>
+  cancel?: () => void
+  name: string
+  uuid: string
+  finished: boolean
+  target?: FileUploadTarget
 }
 
 /** @public */
-export const uploads: Writable<FileUpload[]> = writable([])
-
-/** @public */
-export function dockFileUpload (target: FileUploadTarget, uppy: Uppy): void {
-  uploads.update((instances) => {
-    instances.push({ target, uppy })
-    return instances
-  })
-
-  uppy.on('complete', () => {
-    undockFileUploadIfCompleted(target, uppy)
-  })
+export const uploads = writable(new Map<string, Writable<Upload>>())
```

Comment:
why do you need store of stores? one `writable(new Map<string, Upload>())` store should be enough

Reply:
indeed, fixed.